### PR TITLE
Implement password input on the command line

### LINKED ISSUE #996: Read password from the command line when pulling images
#### What would you like to be added:

Currently the pull command accepts username and password either using a based64 encoded auth string or the credentials put directly on the command line.

I would like to be able to pass the password without having to write it in some fashion that will store it on disk like, for example, in the shell history.

#### Why is this needed:

It would help avoid credentials spreading when people need to pull images manually.

### PR DESCRIPTION
**Pull request by sgaist**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Rather than writing full credentials on the command line when pulling an image this option allows user to pass a username and get asked for the password as can be done with the ssh command line application.

#### Which issue(s) this PR fixes:

Fixes #996

#### Does this PR introduce a user-facing change?

```release-note
Added: enter password on the command line for pull related commands

A new option has been introduced to allow user to enter the registry password securely on the command line rather than as part of the command itself.

Example:
    crictl pull --username foobar registry.example.com/myimage:mytag
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@haircommander PTAL

File: cmd/crictl/image.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (haircommander) [inline cmd/crictl/image.go:8]: nit: if you end up updating something, can you revert this change?

Comment (haircommander) [reply cmd/crictl/image.go:8]: actually, the boilerplate linter is failing because of this

Comment (sgaist) [reply cmd/crictl/image.go:8]: Strange, it is something that was changed when I used one of `go mod vendor` or `gofmt` but running either of them again after reverting the change does not mention it anymore.

Comment (haircommander) [reply cmd/crictl/image.go]: thanks!
